src/test/regress/expected/failure_create_table.out

```diff
@@ -0,0 +1,513 @@
+CREATE SCHEMA failure_create_table;
+SET search_path TO 'failure_create_table';
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- Create distributed table in 1PC and 2PC,
+-- then check if the transaction can be rollbacked.
+SET citus.shard_replication_factor TO 1;
+SET citus.shard_count to 4;
+CREATE TABLE test_table(id int, value_1 int);
+-- Kill connection before sending query to the worker 
+SELECT citus.mitmproxy('conn.kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table','id');
+ERROR:  connection error: localhost:57640
+DETAIL:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+SELECT count(*) FROM pg_dist_shard;
+ count 
+-------
+     0
+(1 row)
+
+\c - - - :worker_1_port
+SELECT count(*) FROM pg_class WHERE relname LIKE 'test_table%';
+ count 
+-------
+     0
+(1 row)
+
+\c - - - :master_port
+SET search_path TO 'failure_create_table';
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- Now, kill the connection just after transaction is opened on
+-- workers.
+SELECT citus.mitmproxy('conn.onQuery(query="^BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table','id');
```

Comment:
It'd also be interesting to fail during distributed table creation where the table has a foreign key to a reference table. Failing on foreign key creation is the one that is a different thing to test. We don't need to add other tests in that scenario.